### DISCUSSION TITLE
Keyboard Events Plus MdiContainer Support for Enterprise Apps

### DISCUSSION THREAD

**Discussion by hintsofttech** [dotnet/maui#270] [Ideas]
We are planning to upgrade our enterprise applications to upcoming MAUI.

We have tried xamarin.forms app components where we found only mobile oriented bindings 
but **not sufficient for Desktop** level supports....

But for enterprise app developers, 
We need **Keyboard Events Plus MdiContainer Support** 
for futuristic enterprise app.
 
Components once created should run across all devices Mobile And Desktop (UWP) ...

1. MdiContainer & MdiChild for every Page i.e. ContentPage, etc...
2. Keyboard Events like Keydown, Keyup, Keypress etc for every Pages and Components (Entry, Label, Button, GridView etc )

Can we expect the above features in upcoming MAUI ?

Regards-
Sanjeeb

**Comment by mouralabank** [comment]
Any updates about the keyboard?

**Comment by mouralabank** [comment]
Can we have multiple document interface support (MDI) or something similar in MAUI? For Desktop applications is something very important and brings several benefits, flexibility and facilities.

Generally desktop platforms have many tools and every user has a way to assemble their workspace. With MDI the user can open multiple tools, move, minimize, maximize, resize and configure their work area as desired and is more useful in each situation.

**Reply by mouralabank** [reply]
Other benefits of MDI for desktop applications:

Improved organization: MDI allows users to organize multiple documents or windows within a single parent window, making it easier to manage and navigate between them.
More efficient use of screen space: By using MDI, desktop applications can maximize screen real estate by consolidating multiple windows into a single parent window.
Better task management: Users can switch between open documents or windows more easily, as they are all contained within a single parent window.
Reduced clutter: MDI can help reduce desktop clutter by consolidating multiple windows into a single parent window, making it easier to find the document or window you need.
Consistent user experience: By using a common interface for multiple documents or windows, MDI can provide a consistent user experience across different parts of the application.
MDI can help improve productivity and user experience in desktop applications by providing a more efficient and organized way to manage multiple documents or windows
